### PR TITLE
feat(docs): update select page to use new layout

### PR DESCRIPTION
**Pull request by MariaJose**
## What?
Update select page to use new layout

## Why?
Provide better experience to the user when reading the docs.

## Screenshots/Screen Recordings
Screen Shot 2022-02-24 at 3 26 31 PM

### CODE REVIEW

File: packages/docs/pages/select.tsx [hunk 1]
```diff
@@ -1,399 +1,499 @@
...
+        <ContentRoutingTabs
+          id="implementation"
+          routes={[
+            {
+              id: 'basic',
+              title: 'Basic',
+              render: () => (
+                <Fragment key="basic">
+                  <Text>
+                    <Code primary>Select</Code> are typeable inputs with selectable options.
```
Comment (MariaJose) [inline packages/docs/pages/select.tsx:35]: I updated `selectable dropdown items` to `selectable options` is that ok? suggestions?

File: packages/docs/pages/select.tsx [hunk 2]
```diff
@@ -1,399 +1,499 @@
...
+                </Fragment>
+              ),
+            },
+            {
+              id: 'max-height',
+              title: 'Max height',
+              render: () => (
+                <Fragment key="max-height">
+                  <Text>
+                    Once the content is longer than the max-height, the <Code primary>Select</Code> will be scrollable.
```
Comment (MariaJose) [inline packages/docs/pages/select.tsx:166]: I updated `the Dropdown will be scrollable` to `the Select will be scrollable`? Is that ok? other suggestions?

File: packages/docs/pages/select.tsx [hunk 3]
```diff
@@ -1,399 +1,499 @@
...
-            </CodePreview>
-          </Panel>
-          <Panel header="Disabled">
-            <Text>
-              It is possible to disable the entire select component, similar to how you would disable a native HTML
-              select element, by using the <Code primary>disabled</Code> property.
-            </Text>
+      <Panel header="Overview" headerId="overview">
+        <Text>
+          <Code primary>Select</Code> give merchants the ability to make a single selection or multiple selections from
```
Comment (rtalvarez) [inline packages/docs/pages/select.tsx:15]: Are we missing an `s` in "give"? So it kinda reads "Select gives merchants the ability ... "

Comment (MariaJose) [reply packages/docs/pages/select.tsx:15]: ohh yes, good catch!

File: packages/docs/pages/select.tsx (hunk 1 above)
Comment (jorgemoya) [reply packages/docs/pages/select.tsx:35]: Good to me.

File: packages/docs/pages/select.tsx [hunk 4]
```diff
@@ -1,399 +1,499 @@
...
+        <ContentRoutingTabs
+          id="implementation"
+          routes={[
+            {
+              id: 'basic',
+              title: 'Basic',
+              render: () => (
+                <Fragment key="basic">
+                  <Text>
+                    <Code primary>Select</Code> are typeable inputs with selectable options.
```
Comment (jorgemoya) [inline packages/docs/pages/select.tsx:35]:
```suggestion
                    <Code primary>Selects</Code> are typeable inputs with selectable options.
```

File: packages/docs/pages/select.tsx (hunk 2 above)
Comment (jorgemoya) [reply packages/docs/pages/select.tsx:166]: Good to me.

**Review by jorgemoya — COMMENTED** [review]
Looking good!

File: packages/docs/pages/select.tsx (hunk 4 above)
Comment (MariaJose) [reply packages/docs/pages/select.tsx:35]: Good catch, ty!

**Review by rtalvarez — APPROVED** [review]

File: packages/docs/pages/select.tsx [hunk 5]
```diff
@@ -1,399 +1,499 @@
...
+          ]}
+        />
+      </Panel>
 
-      <PageNavigation items={items} />
+      <Panel header="Do's and Don'ts" headerId="guidelines">
+        <GuidelinesTable
+          recommended={[
+            <>
+              Typically used within <NextLink href="/panel">Panels</NextLink>.
```
Comment (bc-andreadao) [inline packages/docs/pages/select.tsx:476]:
```suggestion
              Typically use within <NextLink href="/panel">Panels</NextLink>.
```

File: packages/docs/pages/select.tsx [hunk 6]
```diff
@@ -1,399 +1,499 @@
...
+            </>,
+            'All select options should be within a container.',
```
Comment (bc-andreadao) [inline packages/docs/pages/select.tsx:478]:
```suggestion
            'Create all select options within a container.',
```

File: packages/docs/pages/select.tsx [hunk 7]
```diff
@@ -1,399 +1,499 @@
...
+            <>
+              Be used for a single selection between 4 or more pre-defined <Code primary>options</Code>.
```
Comment (bc-andreadao) [inline packages/docs/pages/select.tsx:480]:
```suggestion
              Use for a single selection between 4 or more pre-defined <Code primary>options</Code>.
```

File: packages/docs/pages/select.tsx [hunk 8]
```diff
@@ -1,399 +1,499 @@
...
+            </>,
+            'Default a selection if possible.',
+            'Placeholder text can be added if additional context is needed.',
```
Comment (bc-andreadao) [inline packages/docs/pages/select.tsx:483]:
```suggestion
            'Add placeholder text if additional context is needed.',
```

File: packages/docs/pages/select.tsx (hunk 5 above)
Comment (chanceaclark) [reply packages/docs/pages/select.tsx:476]: The suggestion is now phrase _wierd_. Could we potentially just drop `Typically`?

Comment (MariaJose) [reply packages/docs/pages/select.tsx]: FYI, I dropped the `Typically` 👍🏼

**Review by chanceaclark — APPROVED** [review]

**Pull request merged**